doc/arch/adr-004-user-roles-workflow.md

```diff
@@ -0,0 +1,78 @@
+# User Roles and Boundary Workflow
+
+## User Roles
+
+This app supports three kinds of users: **Contributors** that can add new boundaries, **Validators** that can approve or reject them, and **Administrators** that can add new users and utilities, modify existing ones, and export boundaries. In addition, Administrators can act as both Contributors and Validators in the app.
+
+## Boundary Workflow
+
+The boundary workflow is demonstrated in the following stage diagram:
+
+```mermaid
+stateDiagram-v2
+    direction LR
+    IR: In Review
+    NR: Needs Revisions
+    [*] --> Draft
+    state Contributor {
+        Draft --> Submitted: Submit
+        NR --> Draft: Respond
+        Submitted --> IR: Review
+    }
+    state Validator {
+        IR --> NR: Request Changes
+        IR --> Approved: Approve
+        Approved --> IR: Unapprove
+    }
+    Approved --> [*]
+```
```

Comment:
It would be nice to get the `Review` and `Respond` descriptions to show up in the Validator and Contributor subgraphs respectively, but from the documentation I could not find a way to do this. I think the text description below and role description above is sufficient to convey who has the ability to do these things.